gh-136157: Optimize asyncio.to_thread to avoid contextvars.copy_context() overhead for empty contexts (#136159)
python/cpython: Misc/NEWS.d/next/Library/2025-07-01-02-19-00.gh-issue-136157.jGs_zV.rst

> Optimized ``asyncio.to_thread`` to avoid unnecessary performance overhead from calling ``contextvars.copy_context().run`` when the context is empty.

Reviewer: ```suggestion
Optimized :func:`asyncio.to_thread``to avoid unnecessary performance overhead from calling :meth:`contextvars.Context.run`` when the context is empty.
```
